src/lib/common/sol-log-internal.h

```diff
@@ -36,7 +36,7 @@
 
 #ifdef SOL_LOG_ENABLED
 #define SOL_LOG_INTERNAL_DECLARE(_var, _name)    \
-    struct sol_log_domain _var = {               \
+    SOL_ATTR_UNUSED struct sol_log_domain _var = { \
```

Comment:
why do we need this attribute in this file? I get it for the generated one, but this one we should keep warnings so people can remove unused stuff.

Or we try to solve cases where a combination of `#ifdef` can declare logs but not actually use them?